### PR TITLE
#223 import second timing file

### LINKED ISSUE #223: Import second timing file
Currently, the import component does not have functionality to import a second timing file. 

Also, the parse service does not yet have this functionality.

### PR DESCRIPTION
**Pull request by Dacaspex**
fixes #223 
#### What have I done
- Extended the `ParserService` and `ImportComponent` to be able to import a second timing file.
- Removed the `FileService` as it was giving major headaches on Firefox for some reason and after some inspection, it did not seem necessary to have. 

#### Testing
Import a second fiming file and see that an error is given (in the console) when the parsing fails, or the user is navigated to the workspace view when parsing succeeds

#### Addresses SRD section(s)
- SRD section about `ParserService`: One function has changed completely

**Note:** the low code coverage is due to #222

### CODE REVIEW

**Comment by cwognum** [comment]
I just love how @SquarePlayn changed the title once already and there still is a typo in there!

File: src/app/backend/file-parser/parser.service.ts [hunk 1]
```diff
@@ -66,22 +107,30 @@ export class ParserService {
      * Internal function to create an array of promises that read files
      * @param files Array of files to read
      */
-    private createFileReadPromises(files: File[]): Promise<string>[] {
+    private createFileReadPromiseList(files: File[]): Promise<string>[] {
```
Comment (TijnBertens) [inline src/app/backend/file-parser/parser.service.ts]: Since the code in ```parse()``` and ```parseWithExtraTimingFile``` depend explicitly on the order of elements returned by this function:
```
        // The order of "contents" will have the same order as defined
        // in this array
```
It may be wise to add this to the contract of this function.

File: src/app/import/import.component.ts [hunk 2]
```diff
@@ -114,13 +109,7 @@ export class ImportComponent implements OnInit {
...
-        this.pathFile = this.fileStorage.getFiles().pathFile;
-        this.timingFileOne = this.fileStorage.getFiles().timingFileOne;
-        this.trackFile = this.fileStorage.getFiles().trackFile;
-        this.modelFile = this.fileStorage.getFiles().modelFile;
-        this.timingFileTwo = this.fileStorage.getFiles().timingFileTwo;
-        this.errorFile = this.fileStorage.getFiles().errorFile;
+        // Intentionally left blank
     }
 
     /**
```
Comment (TijnBertens) [inline src/app/import/import.component.ts]: Not part of this PR, but the contracts of the next three functions seem to have been copied without being edited. Can you fix this in this PR?

File: src/app/import/import.component.ts [hunk 3]
```diff
@@ -266,31 +255,42 @@ export class ImportComponent implements OnInit {
      */
     importManually() {
         if (!this.canImport()) {
-            // 'Cannot import manually; not all necessary files are present'
+            // Cannot import manually; not all necessary files are present
```
Comment (nolmoonen) [inline src/app/import/import.component.ts]: Maybe it is better to put this in the error message?

File: src/app/import/import.component.ts [hunk 4]
```diff
@@ -266,31 +255,42 @@ export class ImportComponent implements OnInit {
...
+
+        // Handler when parsing fails
+        const onFailure = (error: Error) => {
+            console.error('An error occurred during parsing');
+            console.error(error);
+        };
+
+        // Handler when parsing succeeds
+        const onSuccess = () => {
+            this.router.navigate(['/']);
```
Comment (TijnBertens) [inline src/app/import/import.component.ts]: Can you add a comment like:
```// Navigate to home if parsing is successful```
Maybe my fault, but I didn't know what this constant ```'/'``` meant.

File: src/app/import/import.component.ts [hunk 5]
```diff
@@ -266,31 +255,42 @@ export class ImportComponent implements OnInit {
...
-            },
-            (error) => {
-                console.log('error');
-                console.log(error);
-            }
-        );
+        const pathFile = this.pathFile.item(0);
+        const timingFileOne = this.timingFileOne.item(0);
+        const trackFile = this.trackFile.item(0);
+        const timingFileTwo = this.timingFileTwo !== undefined
```
Comment (nolmoonen) [inline src/app/import/import.component.ts]: If you break up the ternary operator, you might as well just make this an if-statement.

File: src/app/import/import.component.ts [hunk 6]
```diff
@@ -266,31 +255,42 @@ export class ImportComponent implements OnInit {
...
+            ? this.timingFileTwo.item(0)
+            : undefined;
+
+        // Handler when parsing fails
+        const onFailure = (error: Error) => {
+            console.error('An error occurred during parsing');
```
Comment (nolmoonen) [inline src/app/import/import.component.ts]: I think this is redundant. The message of the thrown error should be sufficient.

File: src/app/import/import.component.ts (hunk 5 above)
Comment (nolmoonen) [reply src/app/import/import.component.ts]: Which is more common and understandable to more people.

File: src/app/backend/file-parser/parser.service.ts (hunk 1 above)
Comment (nolmoonen) [inline src/app/backend/file-parser/parser.service.ts]: What does this breaking up add? This will never be reused and makes the file longer and less clear.

File: src/app/backend/file-parser/parser.service.ts [hunk 7]
```diff
@@ -54,10 +54,51 @@ export class ParserService {
...
+     * Parses the path, timing and track file asynchronously into the data representation.
+     *
+     * @param pathFile Path file
+     * @param timingFile Timing file
+     * @param extraTimingFile Extra timing file
+     * @param trackFile Track file
+     * @param onSuccess Callback upon success
+     * @param onFailure Callback upon failure, receives the error why it fails
+     */
+    parseWithExtraTimingFile(
```
Comment (TijnBertens) [inline src/app/backend/file-parser/parser.service.ts]: This function is not called in any test case, is this because of the bug mentioned in the spec:
```
        // Cannot test functionality properly because of a bug with FileReader
        // See https://github.com/angular/zone.js/issues/1120
        // The FileReader does not work properly with Angular fake async time
```

If so, then for the sake of consistency/coverage, since there is a test case for ```parse()```, shouldn't there also be one for ```parseWithExtraTimingFile()```. Even if the functionality can not properly be tested, the aforementioned comment would still apply and explain why exactly this function is not tested.

File: src/app/import/import.component.ts (hunk 6 above)
Comment (Dacaspex) [reply src/app/import/import.component.ts]: The way JavaScript error tracing works, I think it is beneficial to add these kind of messages. It becomes immediately clear where the error happened (as this might be an error that is thrown because of malformed user input)

File: src/app/backend/file-parser/parser.service.ts (hunk 1 above)
Comment (Dacaspex) [reply src/app/backend/file-parser/parser.service.ts]: Maintainability and SRP mostly

File: src/app/import/import.component.ts (hunk 3 above)
Comment (Dacaspex) [reply src/app/import/import.component.ts]: Fixed

File: src/app/import/import.component.ts (hunk 5 above)
Comment (Dacaspex) [reply src/app/import/import.component.ts]: Fixed, made an if-statement

File: src/app/backend/file-parser/parser.service.ts (hunk 1 above)
Comment (Dacaspex) [reply src/app/backend/file-parser/parser.service.ts]: Merged the two functions

Comment (Dacaspex) [reply src/app/backend/file-parser/parser.service.ts]: Fixed

File: src/app/import/import.component.ts (hunk 2 above)
Comment (Dacaspex) [reply src/app/import/import.component.ts]: Fixed

File: src/app/import/import.component.ts (hunk 4 above)
Comment (Dacaspex) [reply src/app/import/import.component.ts]: Fixed

File: src/app/backend/file-parser/parser.service.ts (hunk 7 above)
Comment (Dacaspex) [reply src/app/backend/file-parser/parser.service.ts]: Fixed, added test case
